File: Data Structures & Algorithms/rotate-array/submission-0.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums.reverse()
        k = k%len(nums)

        nums[:k]= reversed(nums[:k])

        nums[k:]=  reversed(nums[k:])
        



         


        

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {

        unordered_map<int,int> u;

        for(int i : nums){
            u[i]++;
            if(u[i] >1){
                return i;
            }
        }

        return -1;


        
        
    }
};
